webstruct/text_tokenizers.py

```diff
@@ -1,61 +1,139 @@
-    ['Well', ',', 'we', "couldn't", 'have', 'this', 'predictable',
-     ',', 'cliche-ridden', ',', '``', 'Touched', 'by', 'an',
-     'Angel', "''", '(', 'a', 'show', 'creator', 'John', 'Masius',
-     'worked', 'on', ')', 'wanna-be', 'if', 'she', "didn't", '.']
+    >>> WordTokenizer().segment_words(s3)  # doctest: +NORMALIZE_WHITESPACE
+    [TextToken(chars='Well', position=0, length=4),
+     TextToken(chars=',', position=4, length=1),
+     TextToken(chars='we', position=6, length=2),
+     TextToken(chars="couldn't", position=9, length=8),
+     TextToken(chars='have', position=18, length=4),
+     TextToken(chars='this', position=23, length=4),
+     TextToken(chars='predictable', position=28, length=11),
+     TextToken(chars=',', position=39, length=1),
+     TextToken(chars='cliche-ridden', position=41, length=13),
+     TextToken(chars=',', position=54, length=1),
+     TextToken(chars='``', position=56, length=1),
+     TextToken(chars='Touched', position=57, length=7),
+     TextToken(chars='by', position=65, length=2),
+     TextToken(chars='an', position=68, length=2),
+     TextToken(chars='Angel', position=71, length=5),
+     TextToken(chars="''", position=76, length=1),
+     TextToken(chars='(', position=78, length=1),
+     TextToken(chars='a', position=79, length=1),
+     TextToken(chars='show', position=81, length=4),
+     TextToken(chars='creator', position=86, length=7),
+     TextToken(chars='John', position=94, length=4),
+     TextToken(chars='Masius', position=99, length=6),
+     TextToken(chars='worked', position=106, length=6),
+     TextToken(chars='on', position=113, length=2),
+     TextToken(chars=')', position=115, length=1),
+     TextToken(chars='wanna-be', position=117, length=8),
+     TextToken(chars='if', position=126, length=2),
+     TextToken(chars='she', position=129, length=3),
+     TextToken(chars="didn't", position=133, length=6),
+     TextToken(chars='.', position=139, length=1)]
+
+    >>> WordTokenizer().segment_words('"')
+    [TextToken(chars='``', position=0, length=1)]
+
+    >>> WordTokenizer().segment_words('" a')
+    [TextToken(chars='``', position=0, length=1),
+     TextToken(chars='a', position=2, length=1)]
 
     Some issues:
 
-    >>> WordTokenizer().tokenize("Phone:855-349-1914")  # doctest: +SKIP
-    ['Phone', ':', '855-349-1914']
-
-    >>> WordTokenizer().tokenize("Copyright © 2014 Foo Bar and Buzz Spam. All Rights Reserved.")  # doctest: +SKIP
-    ['Copyright', '\xc2\xa9', '2014', 'Wall', 'Decor', 'and', 'Home', 'Accents', '.', 'All', 'Rights', 'Reserved', '.']
-
-    >>> WordTokenizer().tokenize("Powai Campus, Mumbai-400077")  # doctest: +SKIP
-    ['Powai', 'Campus', ',', 'Mumbai", "-", "400077']
-
-    >>> WordTokenizer().tokenize("1 5858/ 1800")  # doctest: +SKIP
-    ['1', '5858', '/', '1800']
-
-    >>> WordTokenizer().tokenize("Saudi Arabia-")  # doctest: +SKIP
-    ['Saudi', 'Arabia', '-']
+    >>> WordTokenizer().segment_words("Phone:855-349-1914")
```

Comment:
I don't think we should enable these tests, they are like `@xfail`

Reply:
So we should remove it at all

Reply:
I think they are nice test cases, so I'd prefer to keep them in some form, maybe convert them to pytest.mark.xfail tests.